example/eblob_backend.cpp

```diff
@@ -66,41 +66,54 @@ int blob_check_corrupted_stamp(void *buffer, size_t buffer_size) {
 		       dc.position == 0;
 	};
 
+	if (!check_dc_header(*dc)) {
+		// eblob_disk_control signature check failed
+		return 0;
+	}
+
+	if (!(dc->flags & BLOB_DISK_CTL_EXTHDR)) {
+		// stop checking: eblob_disk_control signature ok and it's the only header
+		return -EILSEQ;
+	}
+
+	if (buffer_size < sizeof(eblob_disk_control) + sizeof(dnet_ext_list_hdr)) {
+		// dnet_ext_list_hdr is missing
+		return 0;
+	}
+
 	const auto check_ext_header = [] (const dnet_ext_list_hdr &ehdr) {
-		return ehdr.version == DNET_EXT_VERSION_V1 &&
-		       ehdr.timestamp.tsec <= DNET_SERVER_SEND_BUGFIX_TIMESTAMP &&
+		return (ehdr.version == DNET_EXT_VERSION_FIRST || ehdr.version == DNET_EXT_VERSION_V1) &&
+		       ehdr.timestamp.tsec && ehdr.timestamp.tsec <= DNET_SERVER_SEND_BUGFIX_TIMESTAMP &&
 		       !(ehdr.__pad1[0] || ehdr.__pad1[1] || ehdr.__pad1[2]) &&
 		       !(ehdr.__pad2[0] || ehdr.__pad2[1]);
 	};
 
-	if (check_dc_header(*dc) && check_ext_header(*ehdr)) {
+	if (check_ext_header(*ehdr)) {
+		// dnet_ext_list_hdr signature ok
 		return -EILSEQ;
 	}
 
 	return 0;
 }
 
 int blob_read_and_check_stamp(const eblob_backend_config *c,
-                              const dnet_time *timestamp,
+                              const struct dnet_time *timestamp,
```

Comment:
`struct` isn't necessary

Reply:
Oops, I manually undone some changes and copy-pasted it. Thanks.